### PR TITLE
Allow for specifying of brotli via --enable-brotli= or pkg-config

### PR DESCRIPTION
**Pull request by randall**
Addresses issue #2946

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci]
[add to whitelist]

File: configure.ac [hunk 1]
```diff
@@ -1283,11 +1283,77 @@ if test "x${enable_pcre}" != "xyes"; then
   AC_MSG_ERROR([Cannot find pcre library. Configure --with-pcre=DIR])
 fi
 
-# Check Brotli
-AC_CHECK_HEADERS([brotli/encode.h], [has_brotli=1],[has_brotli=0])
-AC_CHECK_LIB([brotlienc],[BrotliEncoderCreateInstance],[AC_SUBST([LIB_BROTLIENC],["-lbrotlienc"])],[has_brotli=0])
-AC_SUBST(has_brotli)
-AM_CONDITIONAL([HAS_BROTLI], [ test "x${has_brotli}" = "x1" ])
+# Brotli check
```
Comment (bryancall) [inline configure.ac]: It would be better to move this to a separate file like build/tcmalloc.m4 is done and then call the function from here.

Comment (zwoop) [reply configure.ac]: Agreed. Make it so #1.

**Comment by zwoop** [comment]
[approve ci]
[add to whitelist]

**Comment by zwoop** [comment]
@randall This might be worth a backport to 7.1.x, if you want to make a PR against the 7.1.x branch?

**Pull request merged**